extras/flake-rust.nix

```diff
@@ -7,18 +7,36 @@ let
   };
   craneLib = crane.lib.${pkgs.system}.overrideToolchain rustWithTools;
 
+  cleanSrc = craneLib.cleanCargoSource (craneLib.path src);
+
   # Library source code with extra dependencies attached
-  fullSrc = pkgs.stdenv.mkDerivation {
-    src = craneLib.cleanCargoSource (craneLib.path src);
-    name = "lbf-rust-build-env";
-    unpackPhase = ''
-      mkdir $out
-      cp -r $src/* $out
-      cd $out
-      ${copyExtraSources}
-      ${copyData}
-    '';
-  };
+  fullSrc =
+    pkgs.stdenv.mkDerivation
+      {
+        src = cleanSrc;
+        name = "${crateName}-build-env";
+        unpackPhase = ''
+          mkdir $out
+          cp -r $src/* $out
+          cd $out
+          ${copyExtraSources}
+          ${copyData}
+        '';
+      };
+
+  vendoredSrc =
+    pkgs.stdenv.mkDerivation
+      {
+        src = cleanSrc;
+        name = "${crateName}-vendored-src";
```

Comment:
how's it different from build-env?

Reply:
This was the result of a long experimentation. 
There's a command called `cargo vendor` which will download all deps and put them into a local directory. Crane also uses this, and I thought I might be able to inject our libraries (such as `lbr-prelude`) into this vendored directory. However I bumped into a few issues, so went back to a simpler solution. However, I still have the feeling that this could be achieved.
Now, this will only grab the source and modify the Cargo.toml such that extra-sources point to `../` instead of `.extras`

Reply:
Also, vendored sources won't include an .extras folder, expecting that it is itself inside one

Reply:
I'll document this